fastplotlib/utils/gui.py

```diff
@@ -0,0 +1,56 @@
+import sys
+import importlib
+
+
+def _is_jupyter():
+    """Determine whether the user is executing in a Jupyter Notebook / Lab."""
+    try:
+        ip = get_ipython()
+        if ip.has_trait("kernel"):
+            return True
+        else:
+            return False
+    except NameError:
+        return False
+
+
+IS_JUPYTER = _is_jupyter()
+
+# Triage. Ultimately, we let wgpu-py decide, but we can prime things a bit to
+# create our own preferred order. If we're in Jupyter, wgpu will prefer and
+# select Jupyter, so no no action needed. If one of the GUI backends is already
+# imported, this is likely because the user want to use that one, so we should
+# honor that. Otherwise, we try importing the GUI backends in the order that
+# fastplotlib prefers. When wgpu-py loads the gui.auto, it will see that the
+# respective lib has been imported and then prefer the corresponding backend.
+
+# A list of wgpu GUI backend libs, in the order preferred by fpl
+gui_backend_libs = ["PySide6", "PyQt6", "PySide2", "PyQt5", "glfw"]
+
```

Comment:
Unless I missed it, we also need a check for if it's running in ipython/jupyter and if `%gui qt` is being used, I found that [this](https://github.com/fastplotlib/fastplotlib/pull/431/files#diff-4fb24c475c133664d23618fd061b4ae9c63fd610f0cf57962a1abfe52bae9ef1L41-L42) worked to do that determination.

I tried to ask here but maybe it's the wrong place to ask :laughing: https://discourse.jupyter.org/t/what-is-the-right-way-to-determine-if-a-gui-qt-event-loop-is-running/22160

Reply:
Also, when the user has invoked `%gui qt` we must use Qt because `jupyter_rfb` stops working in that case, even if you force `canvas = jupyter`.

Reply:
Oh interesting! I had assumed (also in wgpu-py) that if you're in Jupyter you can assume an asyncio event loop and ipywidets to be alove 🤔

Reply:
Can we assume that the `%gui  something`  is called once, before fpl/wgpu is imported, and not changed halfway?

Reply:
I mean, surely a user can do so, but is it reasonable to expect users to expect things to break then 😉 ?

Reply:
> Can we assume that the `%gui something`  is called once, before fpl/wgpu is imported, and not changed halfway?

Yes, changing it in the middle leads to very weird behavior. When they open a nb they should choose jupyter (default) or gui Qt before making any plot.